### PR TITLE
feat: Free storage for not existing files

### PR DESCRIPTION
**Pull request by GNITOAHC**
Add support to clean unused storage for not existing files.
```lua
cmd("Clean", function()
  local storage_path = require("avante.path").get_storage_path()
  for _, file in ipairs(vim.fn.readdir(storage_path)) do
    if file:match("%.json$") then
      local sourcefile = require("avante.path").get_path_from_filename(file)
      if not vim.loop.fs_stat(sourcefile) then
        vim.notify("Deleting " .. storage_path .. "/" .. file)
        vim.fn.delete(storage_path .. "/" .. file)
      end
    end
  end
end, { desc = "avante: clean up" })
```

### CODE REVIEW

File: lua/avante/init.lua [hunk 1]
```diff
@@ -67,6 +67,18 @@ H.commands = function()
       return vim.tbl_filter(function(key) return key:find(prefix, 1, true) == 1 end, Config.providers)
     end,
   })
+  cmd("Clean", function()
```
Comment (aarnphm) [inline lua/avante/init.lua:70]: I think clean should just remove everything

File: lua/avante/path.lua [hunk 2]
```diff
@@ -140,4 +148,8 @@ P.available = function() return templates ~= nil end
 
 P.clear = function() P.cache_path:rm({ recursive = true }) end
 
+P.get_path_from_filename = H.path_from_filename
```
Comment (aarnphm) [inline lua/avante/path.lua:151]: The how purpose of H class (helper class) is to not expose it outside.

File: lua/avante/init.lua (hunk 1 above)
Comment (aarnphm) [reply lua/avante/init.lua:70]: also I think this should just call `require("avante.path").clear()` and the logic should reside there

Comment (GNITOAHC) [reply lua/avante/init.lua:70]: > I think clean should just remove everything

How about a separate command that cleans none existing files?

Comment (aarnphm) [reply lua/avante/init.lua:70]: wdym by none existing files? all files under storage path are json file for chat history.
So it is either just remove all of them or not remove it?

If you want to just clear the chat then use `/clear` in the input buffer.

Comment (GNITOAHC) [reply lua/avante/init.lua:70]: If someone chat with a file `tmp.lua` then remove the file, the chat history will still be stored on the machine. So `Clean` command's purpose is to remove those chat history.

Comment (aarnphm) [reply lua/avante/init.lua:70]: does it matter though? it is going to be like 2kb of json?

If you want to delete the file one can surely just cd to the directory and clear it.

I don't see the point of having this fine grained control

Comment (GNITOAHC) [reply lua/avante/init.lua:70]: Yeah, you're right. I think just remove everything is fine.

File: lua/avante/path.lua [hunk 3]
```diff
@@ -140,4 +140,12 @@ P.available = function() return templates ~= nil end
 
 P.clear = function() P.cache_path:rm({ recursive = true }) end
 
+P.clean = function()
```
Comment (aarnphm) [inline lua/avante/path.lua:143]: just use clear

Comment (GNITOAHC) [reply lua/avante/path.lua]: Sorry for misunderstanding.

File: lua/avante/path.lua [hunk 4]
```diff
@@ -138,6 +138,13 @@ end
...
 P.available = function() return templates ~= nil end
 
-P.clear = function() P.cache_path:rm({ recursive = true }) end
+P.clear = function()
+  P.cache_path:rm({ recursive = true })
+  local storage_path = P.history_path:absolute()
+  for _, file in ipairs(vim.fn.readdir(storage_path)) do
+    if file:match("%.json$") then vim.fn.delete(storage_path .. "/" .. file) end
+  end
+  vim.notify("Clear successful!")
```
Comment (aarnphm) [inline lua/avante/path.lua:147]:
```suggestion
	P.history_path:rm({recursive = true})
```

**Review by aarnphm — APPROVED** [review]
LGTM thanks for going through all the trouble.

**Pull request merged**